lib/connection/connect.js

```diff
@@ -25,13 +28,106 @@ function connect(options, callback) {
           return;
         }
 
-        callback(null, new Connection(ipv4Socket, options));
+        performInitialHandshake(new Connection(ipv4Socket, options), options, callback);
```

Comment:
for DRY purposes:

perhaps:

```
const handshake = (socket) => performInitialHandshake(new Connection(socket, options), options, callback);
```

And then use it where it is necessary.

Reply:
I had considered this, but was also trying to reduce the stack trace. Do you feel strongly about this one? It's only repeated thrice (and only ever will be)

Reply:
My thoughts on this are only that if we ever change this, we have to change it in 3 places. It's not a dealbreaker for me though.